packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -890,19 +897,16 @@ it('should not trigger sorting when filter is clicked', () => {
     isSortable: true,
   });
 
-  setFilter(container, '2');
+  await setFilter(container, '2');
 
   expect(onFilter).toHaveBeenCalled();
   expect(onSort).not.toHaveBeenCalled();
 });
 
-it('should render filter dropdown in the correct document', () => {
-  const mockDocument = new DOMParser().parseFromString(
-    `<!DOCTYPE html><html><body><div></div></body></html>`,
-    'text/html',
-  );
-  const mockContainer = mockDocument.querySelector('div') as HTMLDivElement;
-
+it('should render filter dropdown in the correct document', async () => {
+  const mockDocument = document.implementation.createHTMLDocument();
```

Comment:
It was failing for me before... This helped. I have verified it by getting rid of `ownerDocument` in filter and this test started to fail.